src/jit/lower.cpp

```diff
@@ -372,17 +372,16 @@ void Lowering::DecomposeNode(GenTreePtr* pTree, Compiler::fgWalkData* data)
         }
         break;
     case GT_CALL:
-        NYI("Call with TYP_LONG return value");
         break;
     case GT_RETURN:
-        assert(tree->gtOp.gtOp1->OperGet() == GT_LONG);
+        assert(tree->gtOp.gtOp1->OperGet() == GT_LONG || tree->gtOp.gtOp1->OperGet() == GT_CALL);
         break;
     case GT_STOREIND:
-        assert(tree->gtOp.gtOp2->OperGet() == GT_LONG);
+        assert(tree->gtOp.gtOp2->OperGet() == GT_LONG || tree->gtOp.gtOp2->OperGet() == GT_CALL);
         NYI("StoreInd of of TYP_LONG");
         break;
     case GT_STORE_LCL_FLD:
-        assert(tree->gtOp.gtOp1->OperGet() == GT_LONG);
+        assert(tree->gtOp.gtOp1->OperGet() == GT_LONG || tree->gtOp.gtOp1->OperGet() == GT_CALL);
```

Comment:
To avoid codegen complexity to special case multi-reg call operand for GT_RETURN, GT_STOREIND and GT_STORE_LCL_FLD, as a first cut implementation, how about always forcing IR to be of the form

var=call 

for long returning call nodes so that GT_RETURN, GT_STOREIND and GT_STORE_LCL_FLD doesn't have to worry about seeing a multi-reg returning call as operand.